sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala

```diff
@@ -816,6 +818,15 @@ class JDBCSuite extends QueryTest
     }
   }
 
+  test("register dialect specific functions") {
```

Comment:
This test does not reflect the expectation. The user story should be: if an end-user registers a JDBC catalog with a certain dialect, he/she can directly call functions like `SELECT myCatalog.database.funcName(...)` if the function is registered by the dialect.

The workflow should be
1. JDBC dialect report the functions it supports
2. Spark (JDBCTableCatalog) registers these function
3. end-users call these functions in their queries